### PR TITLE
Add first CRUD operation for User

### PR DESCRIPTION
**Pull request by pastushokk97**

### CODE REVIEW

File: back-end/docker-compose.yml [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
             POSTGRES_PASSWORD: prodpassword
             POSTGRES_USER: postgres
         networks:
             - shared_bridge
+        volumes:
+            - "pg-data:/var/lib/postgresql/data"
+volumes: 
+    pg-data:
+        driver: local
+    app-data:
```
Comment (dhasasoftserve) [inline back-end/docker-compose.yml:31]: Not used volume

File: back-end/package.json [hunk 2]
```diff
@@ -55,7 +57,7 @@
       "ts"
     ],
     "rootDir": "src",
-    "testRegex": ".*\\.spec\\.ts$",
+    "testRegex": ".*\\.test\\.ts$",
```
Comment (dhasasoftserve) [inline back-end/package.json:60]: Good practice to name tests as **spec**

File: back-end/src/fixtures/users.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+export const users = [
+    {
+        email: 'portugal@gmail.com',
+        password: '12345678',
+        countryCode: '351',
```
Comment (dhasasoftserve) [inline back-end/src/fixtures/users.ts:5]: Country code should be in ISO format (2 or 3 chars)
https://countrycode.org/

File: back-end/src/bet/bet.controller.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    constructor(private betService: BetService) {
+    }
+
+    @Get()
+    async getInfo(@Query() query: IdDTO, @Res() res: Response) {
+      const information = await this.betService.getInfo(query.betId);
+    
+      return information ?
+        res.status(HttpStatus.OK).json(information) :
+        res.status(HttpStatus.NOT_FOUND);
```
Comment (dhasasoftserve) [inline back-end/src/bet/bet.controller.ts:17]: You need to send something as a response after the status setting

File: back-end/src/bet/bet.service.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  ) {}
+
+  async registerBet(bet: IBet): Promise<Bet> {
+    console.log(1);
+    const isExist = await this.BetRepository.findOne({
+      userId: bet.userId,
+      competitionId: bet.competitionId
+    });
+
+    if (isExist) {
```
Comment (dhasasoftserve) [inline back-end/src/bet/bet.service.ts:34]: Would be better to use unique index and handle DB error on inserting

File: back-end/src/bet/bet.controller.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { BetService } from "./bet.service";
+import { IdDTO, BetDTO } from './dto/bet.dto'
+
+@Controller('bet')
+export class BetController {
+    constructor(private betService: BetService) {
+    }
+
+    @Get()
+    async getInfo(@Query() query: IdDTO, @Res() res: Response) {
```
Comment (dhasasoftserve) [inline back-end/src/bet/bet.controller.ts:12]: You likely need to get all bets per competition or user than one particular bet

File: back-end/src/user/tests/user.controller.spec.ts [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+  afterEach(async () => {
+    await userRepository.query('DELETE FROM users;');
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe('SUCCESS CASE', () => {
```
Comment (dhasasoftserve) [inline back-end/src/user/tests/user.controller.spec.ts:48]: Would be better to structurize test suites by grouping them by endpoint with positive and negative cases. It gives a more clear vision of particular code block

File: back-end/src/bet/bet.controller.ts (hunk 6 above)
Comment (pastushokk97) [reply back-end/src/bet/bet.controller.ts:12]: Get all user's bets

File: back-end/src/bet/bet.service.ts (hunk 5 above)
Comment (pastushokk97) [reply back-end/src/bet/bet.service.ts:34]: Create an index for userId and competitionId

File: back-end/src/user/tests/user.controller.spec.ts (hunk 7 above)
Comment (pastushokk97) [reply back-end/src/user/tests/user.controller.spec.ts:48]: 1.Rename to the positive and negative cases
2. Rename to SHOULD and SHOULD NOT

**Pull request merged**